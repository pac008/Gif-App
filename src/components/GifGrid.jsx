import { useFetchGifs } from "../hooks"
import { GifItem } from "./GifItem"
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

    const {images, isLoading} = useFetchGifs(category)

  return (
    <>
        
        <h3>{category}</h3>
        {
            isLoading ?
            <h4>Loading...</h4>
            :
            <div className="card-grid">
                { images.map((image) => (<GifItem  key={image.id} {...image} />))}
            </div>
        }

    </>
  )
}


GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}
