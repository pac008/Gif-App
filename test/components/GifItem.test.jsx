import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Test in GifItem', () => { 
    const title = 'test';
    const url = 'http://localhost/123';
    test('Snapshop del componente', () => {
        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })
    
    test('Debe de mostrar la imagen con la url y el ALT indicado', () => {
        render(<GifItem title={title} url={url} />)
        //screen debug()
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)
        const { src, alt } = screen.getByRole('img');
        expect(alt).toBe(title);
        expect(src).toBe(url);
    })

    test('debe de mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })
 })