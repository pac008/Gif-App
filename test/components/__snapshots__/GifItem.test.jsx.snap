// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifItem Snapshop del componente 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="test"
      src="http://localhost/123"
    />
    <p>
      test
    </p>
  </div>
</div>
`;
